lib/puppet/type/keystore_certificate.rb

```diff
@@ -0,0 +1,62 @@
+Puppet::Type.newtype(:keystore_certificate) do
+  desc 'adds a certificate and private key to a pkcs12 keystore'
+
+  ensurable
+
+  def self.title_patterns
+    [ [ /(.+):(.+)/m, [ [:keystore], [:alias] ] ] ]
+  end
+
+  newparam(:alias, :namevar => true) do
+    desc "The certificate alias used to store inside the keystore"
+  end
+
+  newparam(:keystore, :namevar => true) do
+    desc "Path to the keystore to use or create when importing the certificate"
+    isrequired
+  end
+
+  newproperty(:certificate) do
+    desc "Path to the certificate to add to the keystore"
+
+    def fingerprint(file)
+      provider.fingerprint(file)
+    end
+
+    def should_to_s(newvalue)
+      self.class.format_value_for_display(fingerprint(newvalue))
+    end
+
+    def insync?(is)
+      is == fingerprint(should)
+    end
+  end
+
+  newparam(:private_key) do
+    desc "Path to file containing the keystore password"
+    isrequired
+  end
+
+  newparam(:ca_file) do
+    desc "Path to the CA certificate"
```

Comment:
```suggestion
    desc "Path to the file containing th CA certificate"
```

(to be consistent with the other param docs)